http/events.go

```diff
@@ -163,3 +166,18 @@ func handleEventsSubscribe(core *vault.Core, req *logical.Request) http.Handler
 		}
 	})
 }
+
+// prependNamespacePatterns prepends the request namespace to the namespace patterns,
+// and also adds the request namespace to the list.
+func prependNamespacePatterns(patterns []string, requestNamespace *namespace.Namespace) []string {
+	prepend := strings.Trim(requestNamespace.Path, "/")
+	newPatterns := make([]string, 0, len(patterns)+1)
+	newPatterns = append(newPatterns, prepend)
+	for _, pattern := range patterns {
+		if strings.Trim(strings.TrimSpace(pattern), "/") == "" {
+			continue
+		}
+		newPatterns = append(newPatterns, path.Join(prepend, strings.TrimPrefix(pattern, "/")))
```

Comment:
nit: Doesn't matter either way, but `path.Join` calls `Clean` on the return value which replaces multiple slashes for you.

```suggestion
		newPatterns = append(newPatterns, path.Join(prepend, pattern, "/"))
```